tooling/nargo_fmt/src/formatter/expression.rs

```diff
@@ -895,6 +898,68 @@ impl<'a, 'b> ChunkFormatter<'a, 'b> {
         group
     }
 
+    pub(super) fn format_match_expression(
+        &mut self,
+        match_expression: MatchExpression,
+    ) -> ChunkGroup {
+        let group_tag = self.new_group_tag();
+        let mut group = self.format_match_expression_with_group_tag(match_expression, group_tag);
+        force_if_chunks_to_multiple_lines(&mut group, group_tag);
+        group
+    }
+
+    pub(super) fn format_match_expression_with_group_tag(
+        &mut self,
+        match_expression: MatchExpression,
+        group_tag: GroupTag,
+    ) -> ChunkGroup {
+        let mut group = ChunkGroup::new();
+        group.tag = Some(group_tag);
+        group.force_multiple_lines = true;
+
+        group.text(self.chunk(|formatter| {
+            formatter.write_keyword(Keyword::Match);
+            formatter.write_space();
+        }));
+
+        self.format_expression(match_expression.expression, &mut group);
+        self.skip_comments_and_whitespace_chunk();
```

Comment:
```suggestion
        group.trailing_comment(self.skip_comments_and_whitespace_chunk());
```

Reply:
Otherwise the chunk is lost... I know there are no tests for this but this would be the correct way, and in general the tests for comments (after the initial formatter PR) are added as we find bugs for edge cases, so no tests for now is okay 😊